pkg/addon/controllers/addonconfiguration/graph.go

```diff
@@ -369,15 +369,19 @@ func (n *installStrategyNode) addNode(addon *addonv1alpha1.ManagedClusterAddOn)
 		// TODO we should also filter out the configs which are not supported configs.
 		overrideConfigMapByAddOnConfigs(n.children[addon.Namespace].desiredConfigs, addon.Spec.Configs)
 
-		// copy the spechash from mca status
+		// go through mca status ConfigReferences to copy the specHash
```

Comment:
why not we go through the add.Spec.Configs, find the mapped config ref in the status and update specHash? It seems simpler...

Reply:
yes, that's simpler, modified the code.